src/node_url.cc

```diff
@@ -407,29 +407,66 @@ int64_t ParseIPv4Number(const char* start, const char* end) {
   return strtoll(start, nullptr, R);
 }
 
+// https://url.spec.whatwg.org/#ipv4-number-parser
+bool IsIPv4NumberValid(std::string_view input) {
+  if (input.empty()) {
+    return false;
+  }
+
+  if (input.size() >= 2 && input[0] == '0') {
+    if (input[1] == 'X' || input[1] == 'x') {
+      if (input.size() == 2) {
+        return true;
+      }
+
+      return input.find_first_not_of("0123456789abcdefABCDEF", 2) ==
+             std::string_view::npos;
+
+    } else {
+      if (input.size() == 1) {
+        return true;
+      }
+
+      return input.find_first_not_of("01234567", 1) == std::string_view::npos;
+    }
+  }
+
+  return std::all_of(input.begin(), input.end(), ::isdigit);
+}
+
 // https://url.spec.whatwg.org/#ends-in-a-number-checker
-bool EndsInANumber(const std::string& input) {
-  std::vector<std::string> parts = SplitString(input, '.', false);
+bool EndsInANumber(const std::string_view input) {
+  if (input.empty()) {
+    return false;
+  }
+
+  char delimiter = '.';
+  auto pointer_start = input.begin();
+  auto pointer_end = input.end();
 
-  if (parts.empty()) return false;
+  uint8_t parts_size = std::count(pointer_start, pointer_end, delimiter);
+  ++parts_size;
 
-  if (parts.back().empty()) {
-    if (parts.size() == 1) return false;
-    parts.pop_back();
+  if (input.back() == delimiter) {
+    --pointer_end;
+    --parts_size;
   }
 
-  const std::string& last = parts.back();
+  if (parts_size > 1) {
+    pointer_start +=
+        input.rfind(delimiter, std::distance(pointer_start, pointer_end) - 1) +
+        1;
+  }
 
-  // If last is non-empty and contains only ASCII digits, then return true
-  if (!last.empty() && std::all_of(last.begin(), last.end(), ::isdigit)) {
-    return true;
+  if (std::distance(pointer_start, pointer_end) == 0) {
+    return false;
   }
 
-  const char* last_str = last.c_str();
-  int64_t num = ParseIPv4Number(last_str, last_str + last.size());
-  if (num >= 0) return true;
+  if (std::all_of(pointer_start, pointer_end, ::isdigit)) {
+    return true;
+  }
 
-  return false;
+  return IsIPv4NumberValid(std::string(pointer_start, pointer_end));
```

Comment:
This can also be a `std::string_view`, right?

Reply:
But I'll try a little more : )

Reply:
Do you think it is worth to try to use
`basic_string_view( const CharT* s, size_type count );` ?

Reply:
That is the correct one. String_view(start, size_t(end-start))

Reply:
Yeah, `string_view(start, end)` is C++20-only unfortunately

Reply:
Done!